core/src/main/java/org/kiwix/kiwixmobile/core/JNIInitialiser.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core
+
+import android.content.Context
+import android.util.Log
+import org.kiwix.kiwixlib.JNIKiwix
+import org.kiwix.kiwixmobile.core.utils.TAG_KIWIX
+import java.io.File
+import java.io.FileOutputStream
+import javax.inject.Inject
+
+internal class JNIInitialiser @Inject constructor(context: Context, jniKiwix: JNIKiwix) {
+  init {
+    loadICUData(context)?.let(jniKiwix::setDataDirectory)
+  }
+
+  private fun loadICUData(context: Context): String? {
+    return try {
+      val icuDir = File(context.filesDir, "icu")
+      if (!icuDir.exists()) {
+        icuDir.mkdirs()
+      }
+      val icuFileNames = context.assets.list("icu") ?: emptyArray()
+      for (icuFileName in icuFileNames) {
+        val icuDataFile = File(icuDir, icuFileName)
+        if (!icuDataFile.exists()) {
+          FileOutputStream(icuDataFile).use { outputStream ->
+            context.assets.open("icu/$icuFileName").use { inputStream ->
+              inputStream.copyTo(outputStream, 1024)
+            }
+          }
+        }
+      }
+      icuDir.absolutePath
+    } catch (e: Exception) {
+      Log.w(TAG_KIWIX, "Error copying icu data file", e)
+      // TODO: Consider surfacing to user
+      null
+    }
+  }
+
+  fun init() {
+    //  empty method so class is not reported as unused
```

Comment:
Why can't we move the `init` block to this function?

Reply:
The init block was more expressive and I didn't have to store context/JniKiwix in variables, might be better to just suppress the unused warning and delete the init fun

Reply:
Suppressing the warning and adding a doc string explaining why this class is being highlighted as unused will be great.

Reply:
done